src/App.tsx

```diff
@@ -1,26 +1,281 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+//#region imports
+import React, { useEffect, useRef, useState } from 'react';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { SelectedFilter } from './types/SelectedFilter';
+import { TodoItem } from './components/TodoItem/TodoItem';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+// eslint-disable-next-line
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotification';
+import { TodoList } from './components/Todolist/TodoList';
+//#endregion
+
+const ERROR_DELAY = 3000;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  //#region States
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [creatingTodo, setCreatingTodo] = useState<Todo | null>(null);
+
+  const [query, setQuery] = useState<string>('');
+  const [filter, setFilter] = useState<SelectedFilter>(SelectedFilter.ALL);
+
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [isDisabledInput, setIsDisabledInput] = useState<boolean>(false);
+
+  const [loadingIds, setLoadingIds] = useState<number[]>([0]);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+  //#endregion
+
+  //#region useEffects
+  useEffect(() => {
+    if (!isDisabledInput && inputRef.current && loadingIds) {
+      inputRef.current.focus();
+    }
+  }, [isDisabledInput, loadingIds]);
+
+  useEffect(() => {
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+      });
+  }, []);
+  //#endregion
+
+  //#region serverRequest
+  async function addTodo({ title, userId, completed }: Omit<Todo, 'id'>) {
+    setIsDisabledInput(true);
+
+    setCreatingTodo({
+      id: 0,
+      title: title,
+      userId: todoService.USER_ID,
+      completed: false,
+    });
+
+    try {
+      const newPost = await todoService.addTodo({ title, userId, completed });
+
+      setTodos(prev => [...prev, newPost]);
+      setQuery('');
+    } catch (error) {
+      setErrorMessage('Unable to add a todo');
+      setIsDisabledInput(false);
+      setTimeout(() => setErrorMessage(''), ERROR_DELAY);
+    } finally {
+      setCreatingTodo(null);
+      setIsDisabledInput(false);
+    }
   }
 
+  async function deleteTodo(todoId: number) {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const deleteTodo = async (todoId: number) => {
```